File: api/watch.py
"""
Vercel Cron entry point. Vercel invokes this with GET and, when the
CRON_SECRET env var is set on the project, an `Authorization: Bearer
<CRON_SECRET>` header — verified below so the endpoint can't be triggered
by anyone else.
"""

import os
import sys
from http.server import BaseHTTPRequestHandler

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "src"))

from watch_and_update import run_watch  # noqa: E402


class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        cron_secret = os.environ.get("CRON_SECRET")
        if cron_secret and self.headers.get("Authorization") != f"Bearer {cron_secret}":
            self.send_response(401)
            self.end_headers()
            self.wfile.write(b"Unauthorized")
            return

        try:
            result = run_watch()
            self.send_response(200)
            self.send_header("Content-Type", "text/plain")
            self.end_headers()
            self.wfile.write(result.encode("utf-8"))
        except Exception as e:
            self.send_response(500)
            self.send_header("Content-Type", "text/plain")
            self.end_headers()
            self.wfile.write(str(e).encode("utf-8"))

File: api/watch.js
// Vercel Cron entry point. Vercel invokes this with GET and, when the
// CRON_SECRET env var is set on the project, an `Authorization: Bearer
// <CRON_SECRET>` header. That header is verified below so the endpoint
// can't be triggered by anyone else.
import { runWatch } from "../lib/watchAndUpdate.js";

export default async function handler(req, res) {
  const cronSecret = process.env.CRON_SECRET;
  if (cronSecret && req.headers.authorization !== `Bearer ${cronSecret}`) {
    res.status(401).send("Unauthorized");
    return;
  }

  try {
    const result = await runWatch();
    res.status(200).send(result);
  } catch (err) {
    res.status(500).send(String(err && err.message ? err.message : err));
  }
}
